test/test-integration/src/test/java/org/wso2am/micro/gw/tests/endpoints/ProductionSandboxTestCase.java

```diff
@@ -0,0 +1,143 @@
+
+        Assert.assertNotNull(sandResponse);
+        Assert.assertEquals(sandResponse.getResponseCode(), HttpStatus.SC_OK,"Response code mismatched");
+        Assert.assertEquals(sandResponse.getData(), ResponseConstants.API_SANDBOX_RESPONSE,
+                "Response message mismatch.");
+    }
+
+    @Test(description = "Invoke Sandbox endpoint when sandbox endpoints provided alone")
+    public void invokeSandboxEndpointOnly() throws Exception {
+        Map<String, String> sandHeaders = new HashMap<String, String>();
+        sandHeaders.put(HttpHeaderNames.AUTHORIZATION.toString(), "Bearer " + jwtTokenSand);
+        HttpResponse sandResponse = HttpsClientRequest.doGet(getServiceURLHttps(
+                "/v2/sand/pet/findByStatus") , sandHeaders);
+
+        Assert.assertNotNull(sandResponse, "Sandbox endpoint response should not be null");
+        Assert.assertEquals(sandResponse.getResponseCode(), HttpStatus.SC_OK,"Response code mismatched");
+        Assert.assertEquals(sandResponse.getData(), ResponseConstants.API_SANDBOX_RESPONSE,
+                "Response message mismatch.");
+
+        Map<String, String> prodHeaders = new HashMap<String, String>();
+        prodHeaders.put(HttpHeaderNames.AUTHORIZATION.toString(), "Bearer " + jwtTokenProd);
+        HttpResponse prodResponse = HttpsClientRequest.doGet(getServiceURLHttps(
+                "/v2/sand/pet/findByStatus") , prodHeaders);
+
+        Assert.assertNotNull(prodResponse, "Production endoint response should not be null");
+        Assert.assertEquals(prodResponse.getResponseCode(), HttpStatus.SC_UNAUTHORIZED,"Response code mismatched");
+    }
+
+    @Test(description = "Invoke Production endpoint when production endpoints provided alone")
+    public void invokeProdEndpointOnly() throws Exception {
+        Map<String, String> headers = new HashMap<String, String>();
+        headers.put(HttpHeaderNames.AUTHORIZATION.toString(), "Bearer " + jwtTokenProd);
+        HttpResponse response = HttpsClientRequest.doGet(getServiceURLHttps(
+                "/v2/prod/pet/findByStatus") , headers);
+
+        Assert.assertNotNull(response, "Production endpoint response should not be null");
+        Assert.assertEquals(response.getResponseCode(), HttpStatus.SC_OK,"Response code mismatched");
+        Assert.assertEquals(response.getData(), ResponseConstants.RESPONSE_BODY,
+                "Response message mismatch.");
+
+        Map<String, String> sandHeaders = new HashMap<String, String>();
+        sandHeaders.put(HttpHeaderNames.AUTHORIZATION.toString(), "Bearer " + jwtTokenSand);
+        HttpResponse sandResponse = HttpsClientRequest.doGet(getServiceURLHttps(
+                "/v2/prod/pet/findByStatus"), sandHeaders);
+
+        Assert.assertNotNull(sandResponse, "Sandbox endoint response should not be null");
+        Assert.assertEquals(sandResponse.getResponseCode(), HttpStatus.SC_UNAUTHORIZED,"Response code mismatched");
+    }
+
+}
```

Comment:
SAN validation based on verifyHostName
Gateway Types (Prod-sand, prod only, sand only)
Unsecure https communication.

Reply:
VerifyHostName is added via https://github.com/wso2/product-microgateway/pull/1525/commits/5ccd375c928646a871c20c086e1f43f984028d6f

Reply:
Unsecure HTTPS Upstreams are added via https://github.com/wso2/product-microgateway/pull/1525/commits/c571daa34be2a80e114c4f9633517a3ac5b45d32

Reply:
Created a separate issue to keep track of https://github.com/wso2/product-microgateway/issues/1535

Reply:
change verifyhostname string to regex based one rather than exact match.

Reply:
https://www.envoyproxy.io/docs/envoy/v1.16.2/api-v3/extensions/transport_sockets/tls/v3/common.proto#extensions-transport-sockets-tls-v3-certificatevalidationcontext

The wildcard SAN is accepted when using exact keyword
SAN : *.test.com
HOST: abc.test.com (exact match) works

Tested and verified.